src/mpdec.jl

```diff
@@ -0,0 +1,766 @@
+These methods support adaptive time stepping, using the numerical solution obtained with one correction step less as a lower-order approximation to estimate the error.
+The MPDeC schemes were introduced by Torlo and Öffner (2020) for autonomous conservative production-destruction systems and
+further investigated in Torlo, Öffner and Ranocha (2022).
+
+For nonconservative production–destruction systems we use a straight forward extension
+analogous to [`MPE`](@ref).
+A general discussion of DeC schemes applied to non-autonomous differential equations 
+and using general integration nodes is given by Ong and Spiteri (2020).
+
+The MPDeC methods require the special structure of a
+[`PDSProblem`](@ref) or a [`ConservativePDSProblem`](@ref).
+
+You can optionally choose the linear solver to be used by passing an
+algorithm from [LinearSolve.jl](https://github.com/SciML/LinearSolve.jl)
+as keyword argument `linsolve`.
+You can also choose the parameter `small_constant` which is added to all Patankar-weight denominators
+to avoid divisions by zero. You can pass a value explicitly, otherwise `small_constant` is set to
+`1e-300` in double precision computations or `floatmin` of the floating point type used.
+
+## References
+
+- Davide Torlo and Philipp Öffner.
+  "Arbitrary high-order, conservative and positivity preserving Patankar-type deferred correction schemes."
+  Applied Numerical Mathematics 153 (2020): 15-34.
+  [DOI: 10.1016/j.apnum.2020.01.025](https://doi.org/10.1016/j.apnum.2020.01.025)
+- Davide Torlo, Philipp Öffner, and Hendrik Ranocha.
+  "Issues with positivity-preserving Patankar-type schemes."
+  Applied Numerical Mathematics 182 (2022): 117-147.
+  [DOI: 10.1016/j.apnum.2022.07.014](https://doi.org/10.1016/j.apnum.2022.07.014)
+
+- Benjamin W. Ong and Raymond J. Spiteri.
+  "Deferred Correction Methods for Ordinary Differential Equations."
+  Journal of Scientific Computing 83 (2020): Article 60.
+  [DOI: 10.1007/s10915-020-01235-8](https://doi.org/10.1007/s10915-020-01235-8)
+"""
+struct MPDeC{N, F, T2} <: OrdinaryDiffEqAdaptiveAlgorithm
+    K::Int
+    M::Int
+    nodes::N
+    linsolve::F
+    small_constant_function::T2
+end
+
+function small_constant_function_MPDeC(type)
+    if type == Float64
+        # small_constant is chosen such that 
+        # the testet "Zero initial values" passes.
+        small_constant = 1e-300
+    else
+        small_constant = floatmin(type)
+    end
+    return small_constant
+end
+
+function MPDeC(K::Integer;
+               nodes = :gausslobatto,
+               linsolve = LUFactorization(),
+               small_constant = small_constant_function_MPDeC)
+    if small_constant isa Number
+        small_constant_function = Returns(small_constant)
+    else # assume small_constant isa Function
+        small_constant_function = small_constant
+    end
+
+    if nodes === :lagrange
+        M = K - 1
+    else # :gausslobatto 
```

Comment:
Should we through an informative error if `nodes` is neither `:lagrange` nor `:gausslobatto`?